scriptmodules/supplementary/runcommand/joy2key.py

```diff
@@ -30,6 +31,82 @@
 JS_EVENT_AXIS = 0x02
 JS_EVENT_INIT = 0x80
 
+CONFIG_DIR = '/opt/retropie/configs/'
+RETROARCH_CFG = CONFIG_DIR + 'all/retroarch.cfg'
+
+def ini_get(key, cfg_file):
+    pattern = r'[ |\t]*' + key + r'[ |\t]*=[ |\t]*'
+    value_m = r'"*([^"\|\r]*)"*'
+    value = ''
+    with open(cfg_file, 'r') as ini_file:
+        for line in ini_file:
+            if re.match(pattern, line):
+                value = re.sub(pattern + value_m + '.*\n', r'\1', line)
+                break
+    return value
+
+def get_btn_num(btn, cfg):
+    num = ini_get('input_' + btn + '_btn', cfg)
+    if num: return num
+    num = ini_get('input_player1_' + btn + '_btn', cfg)
+    if num: return num
+    return ''
+
+def get_button_codes(dev_path):
+    js_cfg_dir = CONFIG_DIR + 'all/retroarch-joypads/'
+    js_cfg = ''
+    dev_name = ''
+
+    # getting joystick name
+    for device in Context().list_devices(DEVNAME=dev_path):
+        dev_name_file = device.get('DEVPATH')
+        dev_name_file = '/sys' + os.path.dirname(dev_name_file) + '/name'
+        for line in open(dev_name_file, 'r'):
+            dev_name = line.rstrip('\n')
+            break
+    if not dev_name:
+        return default_button_codes
+
+    # getting retroarch config file for joystick
+    for f in os.listdir(js_cfg_dir):
+        if f.endswith('.cfg'):
+            if ini_get('input_device', js_cfg_dir + f) == dev_name:
+                js_cfg = js_cfg_dir + f
+                break
+    if not js_cfg:
+        js_cfg = RETROARCH_CFG
+
+    # getting configs for buttons A, B, X and Y
+    btn_num = {}
+    biggest_num = 0
+    i = 0
+    for btn in 'a', 'b', 'x', 'y':
+        i += 1
+        if i > len(default_button_codes):
+            break
+        btn_num[btn] = get_btn_num(btn, js_cfg)
+        try:
+            btn_num[btn] = int(btn_num[btn])
+        except ValueError:
+            return default_button_codes
+        if btn_num[btn] > biggest_num:
+            biggest_num = btn_num[btn]
+
+    # building the button codes list
+    btn_codes = [''] * (biggest_num + 1)
+    for i in range(len(default_button_codes)):
+        if   i == 0: btn_codes[btn_num['a']] = default_button_codes[0]
+        elif i == 1: btn_codes[btn_num['b']] = default_button_codes[1]
+        elif i == 2: btn_codes[btn_num['x']] = default_button_codes[2]
+        elif i == 3: btn_codes[btn_num['y']] = default_button_codes[3]
```

Comment:
Wondering if the logic could be simplified here by looping though a list of a b x y, and incrementing i as an index, rather than lots of if/elif (and breaking out of the loop if len(default_button_codes) is reached.